From 18eeec99a1bf4a569abb2cf03199b5c8de47c3e2 Mon Sep 17 00:00:00 2001
From: Eduard Dumitrescul <dveduard@google.com>
Date: Tue, 16 Sep 2025 09:47:49 +0000
Subject: [PATCH] Get all accounts no matter the visibility

Bug: 417988098
Test: manual
Flag: EXEMPT BUGFIX
Change-Id: I51c7253b124ab524b817b17dc0aabbdb5ab12ac7
Merged-In: I51c7253b124ab524b817b17dc0aabbdb5ab12ac7
(cherry picked from commit a8819d3bdc496d3a4b5ea041434ff5d71b1f8c95)
---
 .../DevicePolicyManagerService.java           | 19 +++----------------
 1 file changed, 3 insertions(+), 16 deletions(-)

diff --git a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
index 063fb37a8221..d780e8cbbea2 100644
--- a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
+++ b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
@@ -516,6 +516,7 @@ import com.android.server.LockGuard;
 import com.android.server.SystemServerInitThreadPool;
 import com.android.server.SystemService;
 import com.android.server.SystemServiceManager;
+import com.android.server.accounts.AccountManagerService;
 import com.android.server.devicepolicy.ActiveAdmin.TrustAgentInfo;
 import com.android.server.inputmethod.InputMethodManagerInternal;
 import com.android.server.net.NetworkPolicyManagerInternal;
@@ -22745,22 +22746,8 @@ public class DevicePolicyManagerService extends IDevicePolicyManager.Stub {
     }
 
     private boolean hasAccountsOnAnyUser() {
-        long callingIdentity = Binder.clearCallingIdentity();
-        try {
-            for (UserInfo user : mUserManagerInternal.getUsers(/* excludeDying= */ true)) {
-                AccountManager am = mContext.createContextAsUser(
-                                UserHandle.of(user.id), /* flags= */ 0)
-                        .getSystemService(AccountManager.class);
-                Account[] accounts = am.getAccounts();
-                if (accounts.length != 0) {
-                    return true;
-                }
-            }
-
-            return false;
-        } finally {
-            Binder.restoreCallingIdentity(callingIdentity);
-        }
+        AccountManagerService accountManagerService = AccountManagerService.getSingleton();
+        return accountManagerService.getAllAccountsForSystemProcess().length != 0;
     }
 
     private void setBypassDevicePolicyManagementRoleQualificationStateInternal(
-- 
2.52.0.rc1.455.g30608eb744-goog

